server/index.ts

```diff
@@ -56,16 +57,32 @@ export const configSchema = schema.object({
     keepalive: schema.boolean({ defaultValue: true }),
   }),
   auth: schema.object({
-    type: schema.string({
-      defaultValue: '',
-      validate(value) {
-        if (
-          !['', 'basicauth', 'jwt', 'openid', 'saml', 'proxy', 'kerberos', 'proxycache'].includes(
-            value
-          )
-        ) {
-          return `allowed auth.type are ['', 'basicauth', 'jwt', 'openid', 'saml', 'proxy', 'kerberos', 'proxycache']`;
+    type: schema.arrayOf(schema.string(), {
+      defaultValue: [''],
+      validate(value: string[]) {
+        if (!value || value.length === 0) {
+          return `Authentication type is not configurred properly.`;
         }
+
+        if (value.length > 1) {
+          const basicValidate = value.find((element) => {
+            return element.toLowerCase() === 'basicauth';
+          });
+
+          if (!basicValidate) {
+            return `basicauth is mandatory.`;
+          }
+        }
+
+        value.forEach(function (authVal) {
+          if (
+            !['', 'basicauth', 'jwt', 'openid', 'saml', 'proxy', 'kerberos', 'proxycache'].includes(
+              authVal.toLowerCase()
+            )
+          ) {
+            return `allowed auth.type are ['', 'basicauth', 'jwt', 'openid', 'saml', 'proxy', 'kerberos', 'proxycache']`;
+          }
```

Comment:
Potential performance improvement. We should declare authtype array outside:
```
const supportedAuthTypes = ['', 'basicauth', 'jwt', 'openid', 'saml', 'proxy', 'kerberos', 'proxycache'];

const anyUnsupportedAuthTypes = ! (value.every(val => supportedAuthTypes.includes(val.toLowerCase())));

if (anyUnsupportedAuthTypes)
    return `allowed auth types are ['${supportedAuthTypes.join("', '")}']`;

Reply:
Addressed.